In [1]:
from bs4 import BeautifulSoup as bs
from splinter import Browser
import time
import datetime
import pandas as pd
import lxml
import re

In [2]:
df = pd.read_csv("./3-11-23.csv")
df

id                       title     subtitle  \
0      1         Multiples of 3 or 5    Problem 1   
1      2      Even Fibonacci numbers    Problem 2   
2      3        Largest prime factor    Problem 3   
3      4  Largest palindrome product    Problem 4   
4      5           Smallest multiple    Problem 5   
..   ...                         ...          ...   
817  818                         SET  Problem 818   
818  819          Iterative Sampling  Problem 819   
819  820    Nth digit of Reciprocals  Problem 820   
820  821               123-Separable  Problem 821   
821  822         Square the Smallest  Problem 822   

                                               content         release_date  \
0    \nIf we list all the natural numbers below 10 ...  2001-10-05T18:00:00   
1    \nEach new term in the Fibonacci sequence is g...  2001-10-19T18:00:00   
2    \nThe prime factors of 13195 are 5, 7, 13 and ...  2001-11-02T18:00:00   
3    \nA palindromic number reads the same both way...  2001-11-16T18:00:00   
4    \n2520 is the smallest number that can be divi...  2001-11-30T18:00:00   
..                                                 ...                  ...   
817  \n\nThe SET® card game is played with a pack o...  2022-11-27T10:00:00   
818  \nGiven an n-tuple of numbers another n-tuple ...  2022-12-03T13:00:00   
819  \nLet dn(x) be the nth decimal digit of the fr...  2022-12-10T16:00:00   
820  \n\nA set, S, of integers is called 123-separa...  2022-12-17T19:00:00   
821  \n\nA list initially contains the numbers 2,3,...  2022-12-24T22:00:00   

     solved_by_count  difficulty  
0             969832           5  
1             774117           5  
2             557641           5  
3             493738           5  
4             497077           5  
..               ...         ...  
817               93          85  
818              179          45  
819              507          10  
820              105          65  
821              458          15  

[822 rows x 7 columns]

In [47]:
df["content"][56]

'\nIt is possible to show that the square root of two can be expressed as an infinite continued fraction.\n2=1+12+12+12+…\nBy expanding this for the first four iterations, we get:\n1+12=32=1.5\n1+12+12=75=1.4\n1+12+12+12=1712=1.41666…\n1+12+12+12+12=4129=1.41379…\nThe next three expansions are 9970, 239169, and 577408, but the eighth expansion, 1393985, is the first example where the number of digits in the numerator exceeds the number of digits in the denominator.\nIn the first one-thousand expansions, how many fractions contain a numerator with more digits than the denominator?\n'

In [ ]:
# use HTML content as well as text content as columns (html content can be rendered correctly with MathJax in a markdown cell)
# problem 57 example

<p>It is possible to show that the square root of two can be expressed as an infinite continued fraction.</p>
<p class="center">$\sqrt 2 =1+ \frac 1 {2+ \frac 1 {2 +\frac 1 {2+ \dots}}}$</p>
<p>By expanding this for the first four iterations, we get:</p>
<p>$1 + \frac 1 2 = \frac  32 = 1.5$<br />
$1 + \frac 1 {2 + \frac 1 2} = \frac 7 5 = 1.4$<br />
$1 + \frac 1 {2 + \frac 1 {2+\frac 1 2}} = \frac {17}{12} = 1.41666 \dots$<br />
$1 + \frac 1 {2 + \frac 1 {2+\frac 1 {2+\frac 1 2}}} = \frac {41}{29} = 1.41379 \dots$<br /></p>
<p>The next three expansions are $\frac {99}{70}$, $\frac {239}{169}$, and $\frac {577}{408}$, but the eighth expansion, $\frac {1393}{985}$, is the first example where the number of digits in the numerator exceeds the number of digits in the denominator.</p>
<p>In the first one-thousand expansions, how many fractions contain a numerator with more digits than the denominator?</p>